### ISSUE TITLE
Advanced Filtering Updates

### ISSUE THREAD

**Issue by richard-cox** [rancher/dashboard#7206]
This issue description needs updating - but goal is super-charged tables.

https://github.com/rancher/dashboard/issues/7206 should be part of this too.

This is a container issue for items to address before we expand the application of the new [Advanced Filtering](https://github.com/rancher/dashboard/pull/7134) list feature

Functional
- Enhancement - Split out Group By from Show Columns. The show/hide columns feature would be pretty handy in a lot of lists
  - ![image](https://user-images.githubusercontent.com/18697775/196147990-0f6f4611-96fd-4c98-9d8c-e90d19cfdf5f.png)
- Enhancement - Persist show columns settings to be applied over page refresh. 
- Bug - Advanced filtering assumes a columns `getValue` is a function that just returns the property defined in `value`, this is not the case and the function could return a value not related to a single property.
- Investigation - Advanced Filtering may suffer scaling issues, given how often the columns and rows are iterated through. Given changes during the PR though i think these will be similar to those of the normal filter
- Bug - Advanced filtering by labels do not work for Norman resources where the label is not in `metadata.labels` (see comment https://github.com/rancher/dashboard/pull/7134#discussion_r996799448)

UX
- Request - Column filterable warning should be moved to the filter context (dropdown) 
  ![image](https://user-images.githubusercontent.com/18697775/196149306-edbef134-b509-4db9-8895-49b71a39a70d.png)
- Discussion - Discover-ability of the table menu needs improving (i think i recommended this new location, and then forgot and thought the moved group by controls were lost)
- Request - After entering text to filter by hitting return should execute the `add` button
  - ![image](https://user-images.githubusercontent.com/18697775/196149724-d1c6e4fa-0189-4a14-aac6-150a51e90447.png)

**Comment by aalves08** [comment]
Thanks for compiling all of these @richard-cox 🙇 🙇 🙇 🙇

**Comment by aalves08** [comment]
@kwwii , @richard-cox has suggested a number of UX/UI improvements for the advanced filtering feature. Imo, this should have some input from the UX team just to consolidate the experience. What do you think? 🙏

**Comment by kwwii** [comment]
I agree that we should split out the "Group by" functionality. (Should we just remove it for now?)

The 3 dots icon is not enough - even I mistook it for one of the icons in the rows. We should pick a better icon from Material or Eos.

I am not sure about the icon to show which columns are not filterable by. Mainly, I am afraid to start putting icons around things as they seem to add up. I think we need to look into this a bit better and see if there are other alternatives

**Comment by aalves08** [comment]
Other points to be checked/added:
- check sorting of table by labels
- on long label names, hover to display full string

**Comment by gaktive** [comment]
This ticket may be resolved in other ways but moving this to a milestone that's more active and in the ice box to look at later.

**Comment by gaktive** [comment]
We need to circle back to this properly soon though some bit rot has set in here.

**Comment by aalves08** [comment]
@gaktive this may become relevant (not the updates part) because I've just discovered (yesterday) that the pattern used in Elemental https://github.com/rancher/elemental-ui/blob/main/pkg/elemental/pages/_resource/index.vue#L30-L31 breaks the list views. 

Our updates on `master` have made this a buggy view causing the Elemental list views not to show any rows on them 😓 

I'll update Elemental on monday, but I think I'll have to disable the advanced filtering for now (it's only used on Elemental machine inventories view).

FYI @richard-cox

**Comment by richard-cox** [comment]
We need to discuss how this can be tweaked to work with SSP, and how it fits in with Ovi's omni search https://github.com/rancher/dashboard/issues/17747 and the column filters https://github.com/rancher/dashboard/issues/15055

**Comment by aalves08** [comment]
@richard-cox had a chat with Colin recently and he's really interested in the feature of selecting cols to be displayed

**Comment by richard-cox** [comment]
There's some crossover with https://github.com/rancher/dashboard/issues/15222

**Comment by richard-cox** [comment]
Closing given https://github.com/rancher/dashboard/issues/18745
